doc/source/reference/arrays.dtypes.rst

```diff
@@ -328,13 +328,14 @@ Type strings
     The first argument is any object that can be converted into a
     fixed-size data-type object. The second argument is the desired
     shape of this type. If the shape parameter is 1, then the
-    data-type object is equivalent to fixed dtype. If *shape* is a
-    tuple, then the new dtype defines a sub-array of the given shape.
+    data-type object used to be equivalent to fixed dtype. This behaviour is
+    deprecated since NumPy 1.17 and will raise an error in the future.
+    If *shape* is a tuple, then the new dtype defines a sub-array of the given
+    shape.
```

Comment:
I'd lean towards not mentioning what happens if *shape* is 1 at all.

@anntzer, as the author of https://github.com/numpy/numpy/pull/13326, what are your thoughts on mentioning it here?

Reply:
I would hope that after a suitable amount of time has elapsed, one can make 1 equivalent to (1,), but that's basically dependent on the API stability policy which I'm not overly familiar with.